src/search/algorithms.clj

```diff
@@ -46,6 +49,59 @@
                new-came-from
                new-cost-so-far)))))
 
+
+
+
+
+
+(defn heuristic-search [children-fn heuristic-fn start-state is-true & {:keys [max-states] :or {max-states 1000000}}]
+  (loop [frontier (pm/priority-map start-state 0)
+         came-from {}
+         visited #{}]
+    (if (or (empty? frontier)
+            (>= (count came-from) max-states)
+            (is-true (first (peek frontier))))
+      [came-from nil (first (peek frontier))]
+      (let [current (first (peek frontier))
+            children (set (children-fn current))
+            unvisited-children (clojure.set/difference children visited)
+            ;Next line is completely useless
```

Comment:
That is true, and I should have removed it. Actually, you should have removed it as well once you realized that it didn't do anything.